### DISCUSSION TITLE
Transformations workflow: num_samples

### DISCUSSION THREAD

**Discussion by ghost** [Project-MONAI/MONAI#1481] [Q&A] [answered]
Hi,
As part of my transformations workflow I include "RandCropByPosNegLabeld" for my training datasets and included in this I am using num_samples to take multiple crops from the volume.

I want to check that I'm understanding it correctly that for operations after that one, I will now have several cropped samples which are treated independently for following downstream transformations?

Eg:
- If I started with 1 dataset
- Applied RandCropByPosNegLabeld included num_samples = 10, I now have 10 cropped samples
- After this I usually apply Rand3DElasticd, RandAdjustContrastd etc. Are the 10 cropped samples created from the RandCropByPosNegLabeld each treated independently for these operations?

Thanks in advance,
Jeff.

**Comment by Nic-Ma** [comment, marked as answer]
Hi @jnc13 ,

Thanks for your experiments with MONAI.
Yes, you are right, samples are treated independently for following downstream transformations:
https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/utils.py#L372

Thanks.

**Reply by ghost** [reply]
Thanks Nic, just wanted to check.  Keep up the great work!
